pkg/ddl/index.go

```diff
@@ -2563,3 +2565,12 @@ func renameIndexes(tblInfo *model.TableInfo, from, to model.CIStr) {
 		}
 	}
 }
+
+func renameHiddenColumns(tblInfo *model.TableInfo, from, to model.CIStr) {
+	for _, col := range tblInfo.Columns {
+		if getExpressionIndexOriginName(col) == from.O {
+			col.Name.L = strings.Replace(col.Name.L, from.L, to.L, 1)
+			col.Name.O = strings.Replace(col.Name.O, from.O, to.O, 1)
+		}
+	}
+}
```

Comment:
Maybe use `Column.Hidden` to filter hidden columns, otherwise it may cause unexpected behavior if there is a column name contains only one char like `_`, `V` or `$`.